clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -2702,7 +2702,7 @@ Parser::ParseCXXClassMemberDeclaration(AccessSpecifier AS,
   bool MalformedTypeSpec = false;
   if (!TemplateInfo.Kind &&
       Tok.isOneOf(tok::identifier, tok::coloncolon, tok::kw___super)) {
-    if (TryAnnotateCXXScopeToken())
+    if (getLangOpts().CPlusPlus && TryAnnotateCXXScopeToken())
```

Comment:
The caller is named `ParseCXXClassMemberDeclaration` so, I'm not sure it can and should be called outside of C++. Perhaps we should assert instead of checking that is its C++.